### PR TITLE
Switch to git hash for tags by default. 

### PR DESCRIPTION
**Pull request by brendandburns**
Require command line override for something else.

@lavalamp

### CODE REVIEW

**Comment by brendandburns** [comment]
hold on, I want to change the git command...

**Comment by brendandburns** [comment]
Ok, this is now ready for review.

**Comment by thockin** [comment]
LGTM for DNS, the rest seem to follow suit, but i don't want to speak for others too much :)

File: cluster/addons/dns/kube2sky/Makefile [hunk 1]
```diff
@@ -4,7 +4,12 @@
 
 .PHONY: all kube2sky container push clean test
 
-TAG = 1.8
+# By default, pushes to a githash.  To override:
+# make push TAG=<version>
+# Next TAG=1.9
+
+TAG = $(shell git log -1 --format="format:%H")
```
Comment (lavalamp) [inline cluster/addons/dns/kube2sky/Makefile]: Will this include the -dirty marker?

**Comment by lavalamp** [comment]
FWIW, I spent like half an hour fighting with make files to do something similar yesterday, and the syntax you're using totally didn't work on my desktop. Can you also add a "fail if $(TAG) is empty/not set" line to the container/push steps? See https://github.com/GoogleCloudPlatform/kubernetes/blob/master/examples/kubectl-container/Makefile for what I ended up with.

File: cluster/addons/dns/kube2sky/Makefile (hunk 1 above)
Comment (brendandburns) [inline cluster/addons/dns/kube2sky/Makefile]: no, I don't think it does...

I'll switch to your version.

Comment (lavalamp) [inline cluster/addons/dns/kube2sky/Makefile]: My version is unfortunately kubectl specific-- maybe there's an easy way to compute the version string without actually calling one of our binaries-- that would be great, but I didn't find one digging through our shell code...

**Pull request closed without merging**